src/core/components/ChatViewContainer.tsx

```diff
@@ -0,0 +1,65 @@
+import { ChatView } from './ChatView';
+import { ModelConfig, SupportedModels } from '../types/models';
+import type { TextStreamStore } from 'lib-kava-ai';
+import type { ChatCompletionMessageParam } from 'openai/resources';
+import { useMessageHistory } from '../hooks/useMessageHistory';
+import { MessageHistoryStore } from '../stores/messageHistoryStore';
+
+interface ChatViewContainerProps {
+  onMenu: () => void;
+  onPanelOpen: () => void;
+  isPanelOpen: boolean;
+  supportsUpload: boolean;
+  showModelSelector: boolean;
+  startNewChat: () => void;
+  conversationID: string;
+  modelConfig: ModelConfig;
+  errorStore: TextStreamStore;
+  messageStore: TextStreamStore;
+  thinkingStore: TextStreamStore;
+  messageHistoryStore: MessageHistoryStore;
+  isRequesting: boolean;
+  handleChatCompletion: (value: ChatCompletionMessageParam[]) => void;
+  handleCancel: () => void;
+  handleModelChange: (modelName: SupportedModels) => void;
+}
+export const ChatViewContainer = ({
+  onMenu,
+  onPanelOpen,
+  isPanelOpen,
+  supportsUpload,
+  handleModelChange,
+  handleCancel,
+  handleChatCompletion,
+  thinkingStore,
+  messageHistoryStore,
+  isRequesting,
+  errorStore,
+  messageStore,
+  startNewChat,
+  modelConfig,
+  conversationID,
+}: ChatViewContainerProps) => {
+  const { messages } = useMessageHistory(messageHistoryStore);
```

Comment:
Subscribe to changes here so we don't re-render sidebar when new messages are added.